pkg/validation/internal/removed_apis.go

```diff
@@ -403,11 +435,19 @@ func getRemovedAPIsOn1_25From(bundle *manifests.Bundle) (map[string][]string, ma
 					permCheck := func(permField string, perms []v1alpha1.StrategyDeploymentPermissions) {
 						for i, perm := range perms {
 							for j, rule := range perm.Rules {
-								for _, res := range rule.Resources {
-									if _, ok := resInCsvCrds[res]; ok {
-										continue
+								for _, apiGroup := range rule.APIGroups {
+									// if the API Group being used has a potential deprecation
+									// lets check if it is using a deprecated resource as well
+									if apiGroupHasDeprecation(apiGroup) {
+										// check a combination of apiGroup/resource for all
+										// the resources to see if any match the deprecated apis
+										for _, res := range rule.Resources {
+											if _, ok := resInCsvCrds[res]; ok {
```

Comment:
I think that is actually a check that’s saying “if this resource is already covered by one that’s listed in the CSV as something the operator uses, then that’s a more reliable check, so skip this check”

This might be something we want to add a comment for since it isn't immediately obvious why we'd be bailing out when we find an RBAC rule for a resource within a deprecated API.

Reply:
Taking a look at this, I think it could really go either way. The intention behind this section is exactly what @joelanford mentioned.

I think we could either add a comment or adopt the recommendation by @oceanc80 and remove the `continue` statement within the block.

i.e go from:
```go
for _, res := range rule.Resources {
	if _, ok := resInCsvCrds[res]; ok {
		continue
	}
       warnIfDeprecated(schema.GroupResource{Group: apiGroup, Resource: res}, fmt.Sprintf("ClusterServiceVersion.Spec.InstallStrategy.StrategySpec.%s[%d].Rules[%d]", permField, i, j))
}
```
to:
```go
for _, res := range rule.Resources {
	if _, ok := resInCsvCrds[res]; !ok {
            warnIfDeprecated(schema.GroupResource{Group: apiGroup, Resource: res}, fmt.Sprintf("ClusterServiceVersion.Spec.InstallStrategy.StrategySpec.%s[%d].Rules[%d]", permField, i, j))
	}
}
```

Technically saves us a line of code and if it feels more readable that way I am happy to change it